parachain/pallets/assets/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+//! # Assets
+//!
+//! The assets pallet provides functionality for handling account balances.
+//!
+//! ## Overview
+//!
+//! This module is used by the ETH and ERC20 pallets to store account balances for bridged assets.
+//!
+//! Each asset is identified by a unique `H160` hash. This is useful for tracking ERC20 tokens which are natively identified by a 20-byte contract address.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Calls
+//!
+//! - `transfer`: Transferring a balance between accounts.
+//!
+//! ### Public Functions
+//!
+//! - `total_issuance`: Retrieve total issuance for a fungible asset.
+//! - `balance`: Retrieve an account balance.
+//! - `withdraw`: Withdraw from an account.
+//! - `deposit`: Deposit into an account.
+//! - `transfer`: Transfer between accounts.
+//!
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_std::prelude::*;
+use frame_system::{self as system, ensure_signed};
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	traits::Get,
+	dispatch::{DispatchResult, DispatchError},
+};
+
+use sp_core::{U256};
+
+use artemis_core::assets::{AssetId, MultiAsset, SingleAsset};
+use sp_std::marker;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Asset {
+		pub TotalIssuance get(fn total_issuance): map hasher(blake2_128_concat) AssetId => U256;
+		pub Balances get(fn balances): double_map hasher(blake2_128_concat) AssetId, hasher(blake2_128_concat) T::AccountId => U256;
+	}
+	add_extra_genesis {
+		config(balances): Vec<(AssetId, T::AccountId, U256)>;
+		build(|config: &GenesisConfig<T>| {
+			for &(ref asset_id, ref who, amount) in config.balances.iter() {
+				let total_issuance = TotalIssuance::get(asset_id);
+				TotalIssuance::insert(asset_id, total_issuance + amount);
+				Balances::<T>::insert(asset_id, who, amount);
+			}
+		});
+	}
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		<T as system::Trait>::AccountId,
+	{
+		Transferred(AssetId, AccountId, AccountId, U256),
+	}
+);
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		TotalIssuanceOverflow,
+		TotalIssuanceUnderflow,
+		BalanceOverflow,
+		InsufficientBalance
+	}
+}
+
+decl_module! {
+
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		/// Transfer some free balance to another account.
+		// TODO: Calculate weights (#161)
+		#[weight = 10]
+		pub fn transfer(origin,
+						dest: T::AccountId,
+						asset_id: AssetId,
```

Comment:
Can we put `asset_id` as the first argument, as in the other functions?

Reply:
Good catch, will update :+1: